### ISSUE TITLE
Result not as expected

### ISSUE THREAD

**Issue by ShenZheng2000** [mecarill/2pcnet#5]
Hello, authors! 

I've used your provided code to train the model, but I'm having trouble reproducing the results on BDD datasets. The achieved average precision (AP) is much lower than what was reported in the paper. 

I've already requested a pretrained model in #1. Could you please provide the pretrained model or double-check the current code? Your assistance would be greatly appreciated.

This is my result:
![image](https://github.com/mecarill/2pcnet/assets/69662345/a05ab965-4037-482e-a6c1-c1212a4beff0)

This is the result from the paper:
![image](https://github.com/mecarill/2pcnet/assets/69662345/16cdb027-0981-4a76-9bf6-790b50b83b72)

**Update**:
Even after removing the category 'train' during evaluation, the mAP is still ~3% lower.
![image](https://github.com/mecarill/2pcnet/assets/69662345/37250201-b091-4db8-9862-d4312dd700c4)

**Comment by mecarill** [comment]
Hello Shen Zheng! 
Thank you for bringing this up. Our results can be seen below and show the performance as reported. 
<img width="683" alt="Screenshot 2023-05-20 at 3 50 20 PM" src="https://github.com/mecarill/2pcnet/assets/62688286/47d723d8-6f0e-4455-ad5b-3deb2af1d539">

We have updated the readme to include the trained weights for BDD100K and will update the same for SHIFT soon.

**Comment by ShenZheng2000** [comment]
Thanks for your timely response. I have some other questions:

(1) Are you using 3 or 4 GPUs for training? According to the code config file, you are using 4 GPUs. But according to the paper, you are using 3 GPUs. I guess this might cause some discrepencies on the mAP scores. 

(2) Could you provide the groundtruth coco label? I wonder if my category ids used for bdd2coco conversion are correct:
```
        attr_dict["categories"] = [
            {"supercategory": "none", "id": 1, "name": "person"},
            {"supercategory": "none", "id": 2, "name": "rider"},
            {"supercategory": "none", "id": 3, "name": "car"},
            {"supercategory": "none", "id": 4, "name": "truck"},
            {"supercategory": "none", "id": 5, "name": "bus"},
            {"supercategory": "none", "id": 6, "name": "train"},
            {"supercategory": "none", "id": 7, "name": "motor"},
            {"supercategory": "none", "id": 8, "name": "bike"},
            {"supercategory": "none", "id": 9, "name": "traffic light"},
            {"supercategory": "none", "id": 10, "name": "traffic sign"},
        ]     
``` 

(3) Could you check if my training and validation image numbers for BDD day and night is correct?
```
train_clear_daytime: 12454
val_clear_daytime: 1764
train_clear_night: 22884
val_clear_night: 3274
```

Thanks again for your patience, and looking forward to your reply!

**Comment by mecarill** [comment]
(1) Within our paper we use 3 GPUs mainly due to GPU availability. The start-up command is a placeholder for the user to set values, but we will adjust it for clarity. Thanks for your input on this!

(2) We have updated the readme to include our splits under Dataset Download.

(3) Our numbers are higher across the splits. We did not differentiate between clear and other weather conditions and only split between night/day. This is likely the reason for the discrepancy that we have in the evaluation.

`train_night.json: 32998
train_day.json: 36728
val_night.json: 4707
 `

**Comment by ShenZheng2000** [comment]
I appreciate your thorough response! Utilizing the provided checkpoints and JSON files, I successfully reproduced the results described in the paper for the BDD dataset.

However, when I attempt to train the model from scratch, I encounter an unexpected issue where the bounding box mAPs drop to zero during the second evaluation. This situation puzzles me, as the same set of images and the same JSON file are being used for both the first and the second evaluations. Could you provide some clarification on this matter?

![image](https://github.com/mecarill/2pcnet/assets/69662345/9b1cadef-7938-4e07-afc2-c4aaa0de5919)

**Comment by ShenZheng2000** [comment]
I experimented with various torch versions, different dataset splits, and smaller learning rates in an attempt to train the model. However, none of these attempts successfully resolved the issue at hand. On certain occasions, the mean Average Precision (mAP) would turn out to be zero, while at other times, it would become NaN. 

Could you help me on that?

Thanks!

**Comment by mecarill** [comment]
Is this during the burn-up stage? (Before 50k iterations).

If so the teacher has not been initialised yet resulting in 0 AP.

**Comment by ShenZheng2000** [comment]
Yes. It is during the burn-up stage. The mAP after the burn-up stage is good. I will close this issue now.

**Comment by hjw-0909** [comment]
@ShenZheng2000  Why do I report an error every time I run to 50k times?
![image](https://github.com/mecarill/2pcnet/assets/56182057/f2fa4189-6491-4602-bdbb-53189a31130c)

![image](https://github.com/mecarill/2pcnet/assets/56182057/62f9f32d-6845-4b7a-a5a9-d73dee65b586)
